portality/dao.py

```diff
@@ -120,19 +122,29 @@ def save(self, retries=0, back_off_factor=1, differentiate=False, blocking=False
         while attempt <= retries:
             try:
                 r = requests.post(url, data=d)
-                if 400 <= r.status_code < 500:
-                    # bad request, no retry
-                    print "bad request", r.json()
-                    traceback.print_stack(file=sys.stdout)
-                    break
-                elif r.status_code >= 500:
-                    print "server error", r.json()
-                    attempt += 1
+                if r.status_code > 400:
+                    raise ElasticSearchWriteException(u"Error on ES save. Response code {0}".format(r.status_code))
```

Comment:
worth getting the body as well? Not r.json(), but (I assume) r.body or r.text, or whatever the raw response is

Reply:
Oh, my bad, this is handled below